login_server/player/state/authenticated_state.py

```diff
@@ -148,6 +148,25 @@ def handle_server_join_first_step(self, request):
                 return
 
             if game_server.joinable:
+
+                # Check password, if one is necessary
+                if game_server.password_hash is not None:
+                    password_hash_attempt = request.findbytype(m032e)
+                    if password_hash_attempt is None \
+                            or bytes(password_hash_attempt.content) != game_server.password_hash:
+                        # Wrong password / no password
+                        b0msg = a00b0().set_server(game_server).set_player(self.player.unique_id)
+                        b0msg.content.append(m042b().set(STDMSG_INCORRECT_PASSWORD))
+                        self.player.send(b0msg)
+                        self.player.send(a0070().set([
+                            m0348().set(self.player.unique_id),
+                            m0095(),
+                            m009e().set(MESSAGE_UNKNOWNTYPE),
+                            m009d().set(self.player.unique_id),
+                            m02fc().set(STDMSG_INCORRECT_PASSWORD)
+                        ]))
+                        return
```

Comment:
Instead of the return in the middle, I prefer the determination of access denied or granted first, followed by an if statement that has the actions for both cases.

Reply:
Have updated this.